src/uct/cuda/cuda_ipc/cuda_ipc_ep.c

```diff
@@ -22,19 +22,47 @@
 #define UCT_CUDA_IPC_PUT 0
 #define UCT_CUDA_IPC_GET 1
 
+
+static UCS_F_ALWAYS_INLINE void
+uct_cuda_ipc_ep_set_keepalive_device(uct_cuda_ipc_ep_t *ep)
+{
+    if ((ep->keepalive.device == CU_DEVICE_INVALID) &&
+        (UCT_CUDADRV_FUNC(cuCtxGetDevice(&ep->keepalive.device),
+                          UCS_LOG_LEVEL_TRACE) != UCS_OK)) {
```

Comment:
Instead of cuCtxGetDevice, can we use [cuCtxGetCurrent](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__CTX.html#group__CUDA__CTX_1g8f13165846b73750693640fb3e8380d0) ? 
This we directly compare context pointers and avoid the case of application destroying and creating a new context on the same device. If the original context is destroyed then the peer mappings may have to be purged.

Reply:
@dmitrygx Added https://github.com/openucx/ucx/pull/6393 to address the above comment. With #6393, calling  UCT_CUDADRV_GET_CTX(_ctx_ret) instead of specifically looking for driver api ctx or primary ctx should serve the above goal.

Reply:
@Akshay-Venkatesh I see, so the idea is to cache CUDA context instead of CUDA device, right?

Reply:
Right